client/protocols/unlock.py

```diff
@@ -172,6 +203,137 @@ def interact(self, unique_id, case_id, question_prompt, answer, choices=None, ra
     # Private Methods #
     ###################
 
+    # Displaying the correct messages depending on students' wrong answers and treatment group ID
+    def error_handling(self, unique_id, input_lines,access_token):
+
+        shorten_unique_id = unique_id[unique_id.index('\n')+1:]
+
+        # Try to get th info dictionary for this question
+        wa_2_dict_info = json_dict_file['dictAssessId2WA2DictInfo'].get(shorten_unique_id)
+
+        if not wa_2_dict_info: # get returns None if wa_2_dict_info doesn't have shorten_unique_id in dictionary
+            print ("-- Not quite. Try again! --")
+            return ({}, -1)
+
+        dict_info = wa_2_dict_info.get(repr(input_lines))
+
+        # If this wrong answer is not in the JSON file, display default message
+        if not dict_info:            
+            print ("-- Not quite. Try again! --")
+            return ({}, -1)
+
+        # Checks to see the student currently has a treatment group number. If not, calls helper function in auth.py
+        if not os.path.isfile("tests/tg.ok_tg"):
+            cur_email = auth.get_student_email(access_token)
+            if not cur_email:
+                print ("--Not quite. Try again! --")
+                return ({}, -1)
+            lab_assignment = "".join(("".join(self.hash_key.split())).split(":"))
+            try:
+                data = json.loads(urlopen(TGSERVER + cur_email +"/"+lab_assignment+ "/unlock_tg",timeout =1).read().decode("utf-8"))
+            except IOError as e:
+                data = {"tg":-1}
+            fd = open("tests/tg.ok_tg","w")
+            fd.write(str(data["tg"]))
+            fd.close()
+
+        tg_file = open("tests/tg.ok_tg", 'r')
+        tg_id = int(tg_file.read())
+
+        if tg_id == -1: 
+            # If tg_id == -1, some errors happen when trying to access server
+            print ("-- Not quite. Try again! --")
+            return ({}, -1)
+
+        #Get the string that corresponds to this treatment group ID
+        lambda_string = json_dict_file['dictTg2Func'].get(str(tg_id))
+
+        if not lambda_string:
+            print ("-- Not quite. Try again! --")
+            return ({}, -1)
+
+        lambda_info_misu = eval(lambda_string)
```

Comment:
The dictionary in the guidance protocol should be:
```
{
  'none': lambda info,strMisU: None,
  'ki': lambda info,strMisU: info['ki'],
  'reteach': lambda info,strMisU: info['dictMisU2ReteachMsg'][strMisU]
}
```

I just pushed a change to the write_simple_test.py and simple-test.json. Double check that works because that diff seemed to have an extra '\n' in the assessId and I committed but not sure why it had that.